manager/state/raft/transport/peer.go

```diff
@@ -84,31 +108,52 @@ func (p *peer) send(m raftpb.Message) (err error) {
 	return nil
 }
 
-func (p *peer) update(addr string) error {
+func (p *peer) update(addrs ...string) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if p.addr == addr {
-		return nil
+	var (
+		cc   *grpc.ClientConn
+		addr string
+	)
+
+	for _, a := range addrs {
+		c, err := p.tr.dial(a)
+		if err != nil {
+			log.G(p.ctx).WithError(err).Errorf("failed to create conn for %x with addr %s", p.id, a)
+			continue
+		}
+		addr = a
+		cc = c
+		break
 	}
-	cc, err := p.tr.dial(addr)
-	if err != nil {
-		return err
+	if cc == nil {
+		return errors.New("dial failed for all provided addrs")
 	}
 
+	addrMap := make(map[string]struct{}, len(addrs)-1)
+	for _, a := range addrs {
+		if a == addr {
+			continue
+		}
+		addrMap[a] = struct{}{}
+	}
+	p.addrs = addrMap
```

Comment:
We should factor out this code since it looks common between here and `newPeer` (unless I am missing something).